File: CS/Q223568ConditionalFormattingInSL/TestData.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Collections.ObjectModel;

namespace Q223568ConditionalFormattingInSL {
    public class TestData {
        public ObservableCollection<TestDataItem> List { get; private set; }
        public TestData() {
            List = GetTestData();
        }
        ObservableCollection<TestDataItem> GetTestData() {
            Random random = new Random();
            ObservableCollection<TestDataItem> list = new ObservableCollection<TestDataItem>();
            for(int i = 0; i < 10; i++)
                list.Add(new TestDataItem() { Text = "Row" + i % 2, Red = random.Next(255), Green = random.Next(255), Blue = random.Next(255) });
            return list;
        }
    }
    public class TestDataItem {
        public int Red { get; set; }
        public int Green { get; set; }
        public int Blue { get; set; }
        public string Text { get; set; }
    }
}

File: VB/Q223568ConditionalFormattingInSL/TestData.vb
﻿Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports System.Collections.ObjectModel

Namespace Q223568ConditionalFormattingInSL
    Public Class TestData
        Private privateList As ObservableCollection(Of TestDataItem)
        Public Property List() As ObservableCollection(Of TestDataItem)
            Get
                Return privateList
            End Get
            Private Set(ByVal value As ObservableCollection(Of TestDataItem))
                privateList = value
            End Set
        End Property
        Public Sub New()
            List = GetTestData()
        End Sub
        Private Function GetTestData() As ObservableCollection(Of TestDataItem)
            Dim random As New Random()

            Dim list_Renamed As New ObservableCollection(Of TestDataItem)()
            For i As Integer = 0 To 9
                list_Renamed.Add(New TestDataItem() With {.Text = "Row" & i Mod 2, .Red = random.Next(255), .Green = random.Next(255), .Blue = random.Next(255)})
            Next i
            Return list_Renamed
        End Function
    End Class
    Public Class TestDataItem
        Public Property Red() As Integer
        Public Property Green() As Integer
        Public Property Blue() As Integer
        Public Property Text() As String
    End Class
End Namespace
